pkg/bootstrap/bootstrap.go

```diff
@@ -18,83 +19,131 @@ import (
 	"github.com/google/go-containerregistry/pkg/v1/remote"
 	"github.com/google/go-containerregistry/pkg/v1/tarball"
 	errors2 "github.com/pkg/errors"
+	helmv1 "github.com/rancher/helm-controller/pkg/apis/helm.cattle.io/v1"
+	"github.com/rancher/helm-controller/pkg/helm"
 	"github.com/rancher/rke2/pkg/images"
 	"github.com/rancher/wrangler/pkg/merr"
+	"github.com/rancher/wrangler/pkg/schemes"
 	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-var releasePattern = regexp.MustCompile("^v[0-9]")
+var (
+	releasePattern = regexp.MustCompile("^v[0-9]")
+	serializer     = json.NewSerializerWithOptions(json.DefaultMetaFactory, schemes.All, schemes.All, json.SerializerOptions{Yaml: true, Pretty: true, Strict: true})
```

Comment:
How often is this called? Looks like it's only twice in one function so should be defined there so not to pollute package space. If this is a frequent call in the system and better to be defined here for resource purposes, that's fine.

Reply:
It gets used to both decode and encode files from the manifests directory, so it is used more than once but only on startup. For GC purposes I could probably just create it in the calling function and then pass it in.

Reply:
Cool. That would be preferred.